### ISSUE TITLE
ISBAT: Paste a table with Header from Word with the header cells tagged using `<th>`

### ISSUE THREAD

**Issue by bobbingwide** [bobbingwide/bobbingwide#146]
My ex boss and I were discussing accessibility in WordPress. 
In a Word document he'd created there's a table showing a count of WAVE errors.

<img width="914" height="323" alt="Image" src="https://github.com/user-attachments/assets/657a4070-18f7-407f-a7c0-a789461cb3ba" />

I advised him that WordPress has built in functionality to convert a Word table to a table block.
But when I pasted the table into a post it was incorrectly marked up for accessibility.
WAVE produces a _Layout table_ alert.

The pasted HTML was as follows.

```
<figure class="wp-block-table">
<table class="has-fixed-layout">
<thead>
<tr><td>WCAG Error Type</td><td>WordPress Pages with error</td><td>%age of WordPress Pages with errors</td><td>%age of all pages with error</td>
</tr>
</thead>
<tbody>
<tr><td>Low contrast text</td><td>211,092</td><td>87.4%</td><td>79.1%</td></tr><tr><td>Missing alternative text for images</td><td>135,467</td><td>56.1%</td><td>55.5%</td></tr>
<tr><td>Empty links</td><td>123,619</td><td>51.2%</td><td>45.4%</td></tr>
</tbody>
</table>
</figure>
```

### Problem
- The Table settings in the pasted block showed the **Header section** setting toggled on.
- But the cells in the `<tr>` in the `<thead>` were `<td>`'s. 
- They should have been `<th>`'s.

### Notes
- I experimented with a newly created table.

<img width="756" height="334" alt="Image" src="https://github.com/user-attachments/assets/e9f1d7d9-1b71-45a0-9225-1dd3060a7b8c" />

- When I pasted this into the post it was differently marked up.
- The Header section wasn't toggled on.
- This table also got a WAVE Layout table alert.

I don't know what the difference between the two Word tables was and can't explain why the HTML was created differently for them.

**Comment by bobbingwide** [comment]
Post containing the examples is https://s.b/hm/wp-admin/post.php?post=23808&action=edit
The Word source is `bw\a11y\tables.doc`
